discovery_engine_core/web-api/src/bin/ingestion.rs

```diff
@@ -310,78 +335,47 @@ fn serialize_to_ndjson(
 
-#[derive(Debug)]
-struct ElasticOpError;
-impl Reject for ElasticOpError {}
+    /// Upload to elastic did not succeed
+    Elastic(#[source] reqwest::Error),
 
-#[derive(Debug)]
-struct ReceivingOpError;
-impl Reject for ReceivingOpError {}
-
-fn handle_elastic_error(err: reqwest::Error) -> Rejection {
-    error!("ElasticOpError {:#?}", err);
-    warp::reject::custom(ElasticOpError)
+    /// Deserialization of response from elastic instance did not succeed
+    Receiving(#[source] reqwest::Error),
 }
 
-/// An API error serializable to JSON.
-#[derive(Serialize)]
-struct ErrorMessage {
-    code: u16,
-    message: String,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    errored_ids: Option<Vec<String>>,
-}
+impl Reject for Error {}
 
 async fn handle_rejection(err: Rejection) -> Result<impl Reply, Infallible> {
-    let code;
-    let message;
-    let mut errored_ids = None;
-
-    if err.is_not_found() {
-        code = StatusCode::NOT_FOUND;
-        message = "NOT_FOUND";
-    } else if err
-        .find::<warp::filters::body::BodyDeserializeError>()
-        .is_some()
-    {
-        code = StatusCode::BAD_REQUEST;
-        message = "REQUEST_BODY_DESERIALIZATION_ERROR";
-    } else if let Some(ElasticOpError) = err.find() {
-        code = StatusCode::BAD_REQUEST;
-        message = "ELASTIC_ERROR";
-    } else if let Some(ReceivingOpError) = err.find() {
-        code = StatusCode::BAD_REQUEST;
-        message = "RECEIVING_OPERATION_ERROR";
-    } else if let Some(SerializeNdJsonError) = err.find() {
-        code = StatusCode::BAD_REQUEST;
-        message = "NDJSON_SERIALIZATION_ERROR";
-    } else if let Some(EmbeddingsCalculationError(ids)) = err.find() {
-        code = StatusCode::UNPROCESSABLE_ENTITY;
-        message = "UNPROCESSABLE_DOCUMENTS";
-        errored_ids = Some(ids.to_vec());
-    } else {
-        error!("unhandled rejection: {:?}", err);
-        code = StatusCode::INTERNAL_SERVER_ERROR;
-        message = "UNHANDLED_REJECTION";
-    }
-
-    let json = warp::reply::json(&ErrorMessage {
-        code: code.as_u16(),
-        message: message.into(),
-        errored_ids,
-    });
+    let (json, code) = match err.find() {
+        Some(Error::TooManyDocuments) => {
+            let json = reply::json(&json!({}));
+            (json, StatusCode::BAD_REQUEST)
+        }
+        Some(Error::EmbeddingsCalculation(ids)) => {
+            let ingestion_error = IngestionError::new(ids.to_vec());
+            let json = reply::json(&json!(ingestion_error));
+            (json, StatusCode::INTERNAL_SERVER_ERROR)
+        }
+        Some(_) => {
+            let ingestion_error = IngestionError::new(Vec::new());
+            let json = reply::json(&json!(ingestion_error));
+            (json, StatusCode::INTERNAL_SERVER_ERROR)
+        }
+        None => {
+            let json = reply::json(&json!({}));
+            (json, StatusCode::INTERNAL_SERVER_ERROR)
+        }
+    };
```

Comment:
we could also change the return type and then try to make it more unified with user API, and without empty json body:
```rust
async fn handle_rejection(err: Rejection) -> Result<Box<dyn Reply>, Infallible> {
    match err.find() {
        Some(Error::TooManyDocuments) => Ok(Box::new(StatusCode::BAD_REQUEST)),
        Some(Error::EmbeddingsCalculation(ids)) => {
            Ok(Box::new(IngestionError::new(ids.to_vec()).to_reply()))
        }
        Some(_) => Ok(Box::new(IngestionError::new(Vec::new()).to_reply())),
        None => Ok(Box::new(StatusCode::INTERNAL_SERVER_ERROR)),
    }
}
```